pallets/asset-index/src/benchmarking.rs

```diff
@@ -110,39 +99,57 @@ benchmarks! {
 			admin_deposit.into(),
 		));
 
+		let index_tokens = AssetIndex::<T>::index_token_balance(&origin_account_id).into();
 		T::PriceFeedBenchmarks::create_feed(origin_account_id.clone(), asset_id).unwrap();
+		assert_ok!(T::Currency::deposit(asset_id, &origin_account_id, units));
 
 		// construct call
 		let call = Call::<T>::deposit(asset_id, units);
 	}: { call.dispatch_bypass_filter(origin)? } verify {
 		let nav = AssetIndex::<T>::nav().unwrap();
 		let deposit_value = T::PriceFeed::get_price(asset_id).unwrap().checked_mul_int(units.into()).unwrap();
-		let received = nav.reciprocal().unwrap().saturating_mul_int(deposit_value).saturating_add(1u128);
-		assert_eq!(AssetIndex::<T>::index_token_balance(&origin_account_id).into(), received + admin_deposit as u128);
+		let received = nav.reciprocal().unwrap().saturating_mul_int(deposit_value);
+
+		// NOTE:
+		//
+		// the result will be 0 or 1
+		//
+		// - 0 for tests
+		// - 1 for benchmarks ( transaction fee )
```

Comment:
this is rather confusing and depends on the configured fees,
can we simplify this check somehow, by checking that fees are sent to the treasury?
or remove it, this selective check does probably more harm than good